docs/tutorial/2_MarketMaker.md

```diff
@@ -0,0 +1,230 @@
+
+![mew_change](../assets/mew_change.png)
+
+Let's sign and send the transaction. Oh, look at [that](https://ropsten.etherscan.io/tx/0x9e3ef01e47e4a1d6af4d1fbcbcca6f0b7ed287476bddd7a26c174e97ae68788d), I did receive ~0.57 CC in exchange of 1 CLN.
+
+Let's check again the CLN/CC exchange rate, I call `getCurrentPrice` again and get a slightly different price! It returns now 569388278636115886, the delta is 569672901914775677 - 569388278636115886 = 284623278659791. We need to divide it by 1e18, so it gives a change of 0.00028. This means that now we receive less CC for CLN. It's small, but we see that our CC it's getting more expensive. That's the Mechanical Market Maker in action
+
+The most observant of you might noticed that in the last transaction I exchanged 1 CLN for 0.56953055471532534 CC. Let's multiply it by 1e18 to align with other values. 0.56953055471532534 * 1e18 = 569530554715325340, I call this `actualRate`. Then:
+
+> 569672901914775677 > 569530554715325340 > 569388278636115886
+
+Which means:
+
+> getCurrentPrice_before > actualRate > getCurrentPrice_after
+
+I see that actually I exchanged for a slightly different rate than `getCurrentPrice`. Is someone fools me around :monkey_face:?
+
+The exchange rate of CLN/CC is a function with multiple variables, we get to this later. What's important to grasp is that it defined continuously for every wei (smallest indivisible unit) in the domain. So the exchange rate changes with every inserted wei of CLN. When I inserted 1 CLN It was actually 1e18 wei, and every wei got exchanged for an individual price. If it reminds you of something you're right - it's good old calculus. To calculate the `actualRate`, I need to calculate a definite integral of the `getCurrentPrice` function. I stop myself here and refer you to [whitepaper's appendix](https://cln.network/pdf/cln_whitepaper.pdf) for a complete explanation.
+
+
+But actually there's a simple way. I'll explain how it works in the next section, for now we just use the `quote` function. It calculates the exact amount of CC you will get for the specific amount of CLN (and vice versa). Let's say I want to buy 1000 CLN (1e21 wei), firstly I do:
+
+![mew_quote](../assets/mew_quote.png)
+
+It means that now for 1000 CLN I'm getting 471765614363247798186 / 1e18 = ~471.76 CC. Do you remember that in the first part I got 1139 CC for 1000 CLN? That change is pretty rough because there's an edge case when the CC is issued (no CLN have been inserted).
+
+Now after I know for sure how much CC I'll get for 1000 CLN, let's do the trade. But what if between the time when I checked the quote and called `change` function, someone other exchanged CLN to CC before me? I'll get different CC amount! Sometime It can be more (if CC is traded for CLN), but sometimes less. Do I'm ok with less? If yes how much less I'm ok with? Of course this difference is significant only on some edge cases, but we got that covered too.
```

Comment:
* someone other --> someone else
* Sometime It --> Sometimes it
* Do I'm ok with less --> Am I ok with less?